lib/dependabot/metadata_finders/base/changelog_finder.rb

```diff
@@ -344,7 +344,7 @@ def gitlab_client
 
           @gitlab_client ||=
             Gitlab.client(
-              endpoint: "https://gitlab.com/api/v4",
+              endpoint: source.api_endpoint,
```

Comment:
You want to leave this one as `https://gitlab.com/api/v4` - it's used for looking up the changelog of libraries from the hosted gitlab.com instance.

Reply:
Not for this endpoint - it's a nice to have but not required. At some point I'll add the ability for Dependabot to do metadata lookup from self-hosted sources (GitHub Enterprise as well as GitLab), but for now it just does it from the central ones.

Reply:
Ok because I was thinking of adding a factory for the gitlab client to avoid "shotgun surgeries". Maybe we can add two one for the self host and one other for the gitlab.com. What do you think ?

Reply:
Can we get this merged first, so that the script works with self-hosted GitLab instances at all? 

After that, I'm 👍 on adding the ability to lookup changelogs, commit diffs and release notes from self-hosted GitLab instances in theory, although in practice I think it's going to be a pretty rare situation - it will only affect private libraries (anything public is likely to be hosted on a central instance), and even then the only current affect is that they won't have changelog links etc., in the PRs.